tasks.py

```diff
@@ -371,6 +380,32 @@ def get_dependency_version(dependency_name):
     return version_match.group(1)
 
 
+def dump_service_ports_to_disk(context):
+    """Useful for downstream utilities without direct docker access to determine ports."""
+    service_ports = {}
+
+    result = docker_compose(context, "ps --format json", hide=True)
+    for line in result.stdout.splitlines():
+        try:
+            service_def = json.loads(line)
+            service_name = re.search(r"com\.docker\.compose\.service=(?P<service>\w+)", service_def["Labels"]).group(
+                "service"
+            )
+
+            ports_found = {}
+            for port in service_def["Publishers"]:
+                if port.get("PublishedPort", 0) != 0:
```

Comment:
Should be able to just do this:
```suggestion
                if port.get("PublishedPort", 0):
```